pkg/netceptor/tlsconfig_test.go

```diff
@@ -0,0 +1,190 @@
+bnNpYmxlIEF1dG9tYXRpb24gQ29udHJvbGxlciBOb2RlcyBNZXNoIFJPT1QgQ0Ew
+HhcNMjIwMTA3MDAwMzUxWhcNMzIwMTA3MDAwMzUxWjA7MTkwNwYDVQQDEzBBbnNp
+YmxlIEF1dG9tYXRpb24gQ29udHJvbGxlciBOb2RlcyBNZXNoIFJPT1QgQ0EwggIi
+MA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQCxAErOWvVDU8mfZgtE6BSygTWU
+MkPxxIEQSYs/UesRAHaB+QXa7/0Foa0VUJKcWwUE+2yYkNRrg8MmE8VWMSewcaNI
+As407stFXP+A2anPEglwemTskpO72sigiYDKShC5n5ciyPsHckwVlOCTtac5TwFe
+eTmGnHWRcd4uBGvaEXx98fw/wLgYtr9vmKTdnOQjriX9EaAWrjlrlzm54Bs3uVUj
+GSL7zY381EuUVV4AjbqQyThbY9cVfsK0nmzLUqpiHG2IhGZDZA9+jxtz2wJWFkNQ
+nWA3afCUjcWV+4FpP3p1U1myCeh2yR2uCHs9pkUK3ts9uD/Wd5j9M1oBMlymbN/C
+5Fahd+cTXrPAjsoRqCso9TBP4mIlNl1Jq8MRUWTL5HOuwn+KnufBtuQ1hIb71Eso
+kj90eWeo/P+temYAEquUVWiej7lnHyZVW647lE+o+xJEOmW+tY5H4jgA/twP4s7U
+BgR545usWF9/utvnhsGSkg1EYcdzaM01pkrWrw1GvHT++HshsrG6Tse8gY7JrTds
+LDtU8LPhPUEnSfVBcgcMg2Dg8lEbaODtdp2xtCJwZHy9CiAx3CKcogVEfecrKSSr
+2iSocft9/l8J+mhVG2CI6ekG6Cy9hDct/3SV01Dfd4FG7xXJIE3mTDLILpi1AVIW
+MHxYD3skuQMyLhJDAwIDAQABo2EwXzAOBgNVHQ8BAf8EBAMCAoQwHQYDVR0lBBYw
+FAYIKwYBBQUHAwIGCCsGAQUFBwMBMA8GA1UdEwEB/wQFMAMBAf8wHQYDVR0OBBYE
+FFAlC81rH211fPJoWglERKb/7/NfMA0GCSqGSIb3DQEBCwUAA4ICAQCWCP/O6YQ9
+jhae2/BeUeeKsnoxf90prg3o6/QHbelF6yL+MvGg5ZbPSlt+ywLDNR2CYvXk/4SD
+5To7CPKhSPBwwUJpafvQfAOZijU30fvXvp5yZEFoOzOyvBP58NfzL5qH6Pf5A6i3
+rHvtR1v7DgS7u2qWWcSimIM0UPoV3JubLTEORjOR6FIyNkIxdjhrP3SxyZ54xxde
+G3bchKaRcGVNoFYSDN4bAA22JAjlD8kXNYKzIS/0cOR/9SnHd1wMIQ2trx0+TfyG
+FAA1mW1mjzQd+h5SGBVeCz2W2XttNSIfQDndJCsyACxmIaOK99AQxdhZsWfHtGO1
+3TjnyoiHjf8rozJbAVYqrIdB6GDf6fUlxwhUXT0qkgOvvAzjNnLoOBUkE4TWqXHl
+38a+ITDNVzaUlrTd63eexS69V6kHe7mrqjywNQ9EXF9kaVeoNTzRf/ztT/DEVAl+
+rKshMt4IOKQf1ScE+EJe1njpREHV+fa+kYvQB6cRuxW9a8sOSeQNaSL73Zv54elZ
+xffYhMv6yXvVxVnJHEsG3kM/CsvsU364BBd9kDcZbHpjNcDHMu+XxECJjD2atVtu
+FdaOLykGKfMCYVBP+xs97IJO8En/5N9QQwc+N4cfCg9/BWoZKHPbRx/V+57VEj0m
+69EpJXbL15ZQLCPsaIcqJqpK23VyJKc8fA==
+-----END CERTIFICATE-----
+`))
+
+	return tempCertFile, tempCertKey, tempCA, func(t *testing.T) {
+		defer os.Remove(tempCertFile.Name())
+		defer os.Remove(tempCertKey.Name())
+		defer os.Remove(tempCA.Name())
+	}
+}
+
+func TestPrepareGoodTlsServerCfg(t *testing.T) {
+	tempCertFile, tempCertKey, tempCA, teardownSuite := setupSuite(t)
+	defer teardownSuite(t)
+
+	cfg := &tlsServerCfg{
+		Name:              "foobar",
+		Cert:              tempCertFile.Name(),
+		Key:               tempCertKey.Name(),
+		RequireClientCert: false,
+		ClientCAs:         tempCA.Name(),
+	}
+
+	MainInstance = New(context.Background(), "foobar")
+
+	err := cfg.Prepare()
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func TestNodeIDMismatch(t *testing.T) {
+	tempCertFile, tempCertKey, tempCA, teardownSuite := setupSuite(t)
+	defer teardownSuite(t)
+
+	cfg := &tlsServerCfg{
+		Name:              "foobar",
+		Cert:              tempCertFile.Name(),
+		Key:               tempCertKey.Name(),
+		RequireClientCert: false,
+		ClientCAs:         tempCA.Name(),
+	}
+
+	MainInstance = New(context.Background(), "barfoo")
+
+	err := cfg.Prepare()
+	if err == nil {
+		t.Error(err.Error())
```

Comment:
is there a way we can capture and assert on the mismatch error her? that way if an error unrelated to the mismatch creeps up, we'll fail this test.

i.e. if err.Error() != "nodeID=foobar not found in certificate..."